core/src/server/rpc_module.rs

```diff
@@ -366,22 +367,25 @@ impl Methods {
 		subscription_permit: SubscriptionPermit,
 	) -> RawRpcResponse {
 		let (tx, mut rx) = mpsc::channel(buf_size);
-		let Request { id, method, params, mut extensions, .. } = req;
+		// The extensions is always empty when calling the method directly because decoding an JSON-RPC
+		// request doesn't have any extensions.
+		let Request { id, method, params, .. } = req;
 		let params = Params::new(params.as_ref().map(|params| params.as_ref().get()));
 		let max_response_size = usize::MAX;
 		let conn_id = ConnectionId(0);
-		extensions.insert(conn_id);
+		let mut ext = self.extensions.clone();
+		ext.insert(conn_id);
```

Comment:
we could actually inject this dummy connection in the constructor but I don't want to add the overhead for stuff that isn't using it.